netmap/netmap_contract.go

```diff
@@ -623,8 +623,13 @@ func SubscribeForNewEpoch(contract interop.Hash160) {
 
 	ctx := storage.GetContext()
 	var num byte
-	it := storage.Find(ctx, []byte(newEpochSubscribersPrefix), storage.None)
+	it := storage.Find(ctx, []byte(newEpochSubscribersPrefix), storage.KeysOnly|storage.RemovePrefix)
 	for iterator.Next(it) {
+		raw := iterator.Value(it).([]byte)[1:] // 1 byte is an index
+		if contract.Equals(raw) {
+			return
+		}
```

Comment:
I'm a paranoiac about VM's `EQUAL` and compiler, could we have a test for the contract that subscribes for notifications and after that subscribes one more time? In the end we should check that there's only one subscriber record stored in the netmap.